src/core/ext/filters/client_channel/global_subchannel_pool.h

```diff
@@ -57,9 +47,12 @@ class GlobalSubchannelPool final : public SubchannelPoolInterface {
       ABSL_LOCKS_EXCLUDED(mu_);
 
  private:
+  GlobalSubchannelPool() {}
+  ~GlobalSubchannelPool() override {}
+
   // The singleton instance. (It's a pointer to RefCountedPtr so that this
   // non-local static object can be trivially destructible.)
-  static RefCountedPtr<GlobalSubchannelPool>* instance_;
+  static std::atomic<RefCountedPtr<GlobalSubchannelPool>*> instance_;
```

Comment:
I think we can remove this data member and instead make it a local static variable within the `instance()` method.  That way, it does not need to be atomic.

Reply:
It has a non-trivial destructor and so cannot be made static per style guide.

Reply:
We can't have a static object of type `RefCountedPtr<GlobalSubchannelPool>`, but we can have a static pointer to that type.  In other words:

```
static RefCountedPtr<GlobalSubchannelPool> GlobalSubchannelPool::instance() {
  static GlobalSubchannelPool* instance = new GlobalSubchannelPool();
  return instance->Ref();
}
```

The object will never be deleted, but that's probably okay for this kind of thing.

Reply:
Done